java/vector/src/main/codegen/templates/UnionMapWriter.java

```diff
@@ -140,6 +140,33 @@ public UnionMapWriter value() {
     return this;
   }
 
+  private String getWriteFieldName() {
+    Field mapField = this.vector.getField();
+    if (mapField == null) {
+      throw new UnsupportedOperationException("MapVector does not have a field.");
+    }
+    if (mapField.getChildren().size() != 1) {
+      throw new UnsupportedOperationException("MapVector does not have a single struct field.");
+    }
+    Field structField = mapField.getChildren().get(0);
+    switch (mode) {
+      case KEY:
+        if (structField.getChildren().size() == 2) {
+          return structField.getChildren().get(0).getName();
+        } else {
+          return MapVector.KEY_NAME;
```

Comment:
This should only be if there are 0 children? If there's any other number of children presumably it should fail

Reply:
Right, so better to check `size==0` and assign the `KEY_NAME` else, the field from the struct. Also for value, I think the most accurate check would be `size < 2`, because there could be no key defined, then there could be 1 child (when key was added).